Sources/Routing/Routing/TrieRouter.swift

```diff
@@ -85,8 +85,8 @@ public final class TrieRouter<Output> {
                 // if no constant routes were found that match the path, but
                 // a dynamic parameter child was found, we can use it
                 let value = ParameterValue(
-                    slug: String(data: parameter.value, encoding: .utf8) ?? "",
-                    value: path.routerParameterValue
+                    slug: String(data: parameter.value, encoding: .utf8)?.removingPercentEncoding ?? "",
```

Comment:
The slug comes from the route registration (i.e, `router.get("users/:id")`). We shouldn't need to percent decode it.